### PR TITLE
wip on installing es, kibana & GN4

### PR DESCRIPTION
**Pull request by landryb**
work in progress ... to test georchestra/georchestra#3218

### CODE REVIEW

**Comment by landryb** [comment]
as spotted by @pmauduit those bits of config in https://github.com/georchestra/geonetwork/blob/georchestra-gn4-4.0.5/docker/kibana/kibana.yml need to be injected in `/etc/kibana/kibana.yml` which isnt supported by `geerlingguy.kibana` role. once injected manually kibana is available inside geonetwork.

**Comment by pmauduit** [comment]
After my modifications above, I now have a gn4 working in the vagrant env, but the link to kibana does not work yet.

Also Mapstore2 is deployed by default.

File: roles/georchestra/tasks/wars.yml [hunk 1]
```diff
@@ -21,7 +21,41 @@
...
+    src: "/usr/share/lib/georchestra-{{ item.key }}.zip"
+    dest: /usr/share/lib
+    remote_src: yes
+  with_dict: "{{ georchestra_wars }}"
+  when: item.value.enabled|bool == True and item.value.url is defined
+
+- name: creating symlinks for downloaded webapps
+  file:
+    owner: tomcat
+    src: "/usr/share/lib/{{ item.key }}{{ '-' if item.value.artifact_sha256 is defined }}{{ item.value.artifact_sha256 if item.value.artifact_sha256 is defined }}.war"
```
Comment (landryb) [inline roles/georchestra/tasks/wars.yml:46]: that line blows for me on ansible 2.9.16, this works:
```
src: "/usr/share/lib/{{ item.key }}{% if item.value.artifact_sha256 is defined %}-{{ item.value.artifact_sha256  }}{% endif %}.war"
```

Comment (pmauduit) [reply roles/georchestra/tasks/wars.yml:46]: applied locally already

File: roles/georchestra/tasks/wars.yml [hunk 2]
```diff
@@ -21,7 +21,41 @@
...
+    dest: "{{ tomcat_basedir }}/{{ item.value.tomcat }}/webapps/{{ item.key }}.war"
+    state: link
```
Comment (landryb) [inline roles/georchestra/tasks/wars.yml:48]: `force: yes` welcome to be able to upgrade a running webapp pointing the symlink to a new version. This blows for gn4 as it doesnt support reload and requires a tomcat restart, but still better.

File: roles/georchestra/tasks/wars.yml [hunk 3]
```diff
@@ -21,7 +21,41 @@
...
+  with_dict: "{{ georchestra_wars }}"
+  when: item.value.enabled|bool == True and item.value.url is defined
+
+
+- name: removes the downloaded archive
+  file:
+    path: "{{ tomcat_basedir }}/{{ item.value.tomcat }}/webapps/{{ item.key}}.zip"
```
Comment (landryb) [inline roles/georchestra/tasks/wars.yml:55]: `path: "/usr/share/lib/georchestra-{{ item.key }}.zip"` is the right path for cleanup :)

**Comment by landryb** [comment]
+1 on the branch merge once it's rebased on top of master and the nits are fixed :)

File: roles/georchestra/tasks/wars.yml (hunk 2 above)
Comment (pmauduit) [reply roles/georchestra/tasks/wars.yml:48]: done locally, will commit once validated

File: roles/georchestra/tasks/wars.yml (hunk 3 above)
Comment (pmauduit) [reply roles/georchestra/tasks/wars.yml:55]: yup, also done.

**Comment by pmauduit** [comment]
force-pushed.

Note: when fiddling with different versions of ansible, please note that `pip install ansible` does not install the python module named `six`, which is necessary for the role `elastic.elasticsearch`.

**Comment by pmauduit** [comment]
the kibana version which is installed is too recent compared to the elasticsearch one:

```
This version of Kibana (v7.14.0) is incompatible with the following Elasticsearch nodes in your cluster: v7.13.4 @ 127.0.0.1:9200 (127.0.0.1)
```

I guess it should be ok to bump the elasticsearch version.

**Pull request merged**